python/triton/tools/aot.py

```diff
@@ -87,6 +66,38 @@
         if args.features:
             arch_features = args.features
 
+        # triton-ir -> triton-gpu-ir
+        # use compute_capability == 80
+        module = triton.compiler.ttir_to_ttgir(module, num_warps=4, num_stages=3, compute_capability=80)
+        # triton-gpu-ir -> llvm-ir
+        # use compute_capability == 80
+        module = triton.compiler.ttgir_to_llir(module, extern_libs=None, compute_capability=80)
+        # llvm-ir -> amdgcn asm, hsaco binary
         module, hsaco_path = triton.compiler.llir_to_amdgcn_and_hsaco(module, arch_name, arch_triple, arch_features)
 
+        print(hsaco_path)
+        print(module)
+        sys.exit(0)
+
+    if not args.sm:
+        raise argparse.ArgumentError(None, "Must specify --sm for PTX compilation")
+
+    # triton-ir -> triton-gpu-ir
+    module = triton.compiler.ttir_to_ttgir(module, num_warps=4, num_stages=3, compute_capability=args.sm)
+    if args.target == 'triton-gpu-ir':
+        print(module.str())
+        sys.exit(0)
+
+    # triton-gpu-ir -> llvm-ir
+    module = triton.compiler.ttgir_to_llir(module, extern_libs=None, compute_capability=args.sm)
+    if args.target == 'llvm-ir':
+        print(module)
+        sys.exit(0)
+
+    # llvm-ir -> ptx
+    if args.target == 'ptx':
+        if not args.ptx_version:
+            raise argparse.ArgumentError(None, "Must specify --ptx-version for PTX compilation")
+        module = triton.compiler.llir_to_ptx(module, compute_capability=args.sm, ptx_version=args.ptx_version)
+
```

Comment:
This code is same as deleted code above.
I switched amd related code before PTX related code to preserve old PTX related code untouched and make AMD related code independent from PTX.